src/plugins/discover/public/application/view_components/canvas/discover_chart_container.tsx

```diff
@@ -22,7 +22,7 @@ import { DiscoverChart } from '../../components/chart/chart';
 import { QUERY_ENHANCEMENT_ENABLED_SETTING } from '../../../../common';
 import { getNewDiscoverSetting, setNewDiscoverSetting } from '../../components/utils/local_storage';
 
-export const DiscoverChartContainer = ({ hits, bucketInterval, chartData }: SearchData) => {
```

Comment:
is rows and chartData different?

Reply:
This is an example of chart data, which is used to render the discover histogram. 

![Screenshot 2025-03-07 at 4 08 31 PM](https://github.com/user-attachments/assets/60b8a360-3b92-4905-a61d-6ce32434bc72)

Reply:
chart data will only be produced if discover histogram exists --> which means chart data will not be produced by sending any PPL query other than the default one

rows will always contain information(with an aggregated value or not) since this is the information we will output in the discover table. Therefore, i think it makes sense to use the rows data here to render visualizations in discover.

Reply:
I also see histogram in discover is rendered by directing using [<Chart>](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/2a9cb948bf42e7424bdd68e83aca28e7a49e8b2a/src/plugins/discover/public/application/components/chart/histogram/histogram.tsx) instead of using expression renderer. But since for discover visualization we might wish to onboard more visualization type in the future, it might be more beneficial to take the expressions renderer approach.